src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameRequestStream.cs

```diff
@@ -193,6 +198,16 @@ public void Abort(Exception error = null)
             }
         }
 
+        public FrameRequestStream Upgrade()
+        {
+            var upgraded = new FrameRequestStream(this);
+            if (_state != FrameStreamState.Aborted)
+            {
+                _state = FrameStreamState.Upgraded;
```

Comment:
Why not change state only if Aborted, but not closed? Why still return an upgraded stream?